### PR TITLE
Sim tests refactoring

### PR DESCRIPTION
**Pull request by nazarhussain**
**Motivation**

Some pending refactoring, suggested in previous discussions. 

**Description**

Sim tests refactoring based on previous suggestions/feedback. 

**Steps to test or reproduce**

Run all tests.

### CODE REVIEW

File: packages/cli/test/utils/simulation/interfaces.ts [hunk 1]
```diff
@@ -104,8 +108,7 @@ export interface CLNode {
   readonly url: string;
   readonly api: Api;
   readonly keyManager: KeyManagerApi;
-  readonly localKeys: SecretKey[];
-  readonly remoteKeys: SecretKey[];
+  readonly keys: CLClientKeys;
```
Comment (g11tech) [inline packages/cli/test/utils/simulation/interfaces.ts:111]: but there can be both local and remote kets in the CL

Comment (nazarhussain) [reply packages/cli/test/utils/simulation/interfaces.ts:111]: Yes there can be, but as per @dapplion we don't want to make SIM tests complex and mix up the keys. You can skill have local or remote keys in multiple CL nodes.

File: packages/cli/test/utils/simulation/SimulationEnvironment.ts [hunk 2]
```diff
@@ -136,12 +140,16 @@ export class SimulationEnvironment {
...
   }
 
-  async start(timeout: number): Promise<void> {
-    try {
-      setTimeout(async () => {
-        await this.stop(1, "On timeout");
-      }, timeout);
+  async start(opts: StartOpts): Promise<void> {
+    setTimeout(() => {
+      this.stop(1, "Overall run timeout").catch((e) => console.error("Error on stop", e));
```
Comment (dapplion) [inline packages/cli/test/utils/simulation/SimulationEnvironment.ts]: Add timeout amount in ms in the error message, same below at
```
Start timeout, no genesis in ${timeout} ms
```

File: packages/cli/test/utils/simulation/interfaces.ts (hunk 1 above)
Comment (dapplion) [reply packages/cli/test/utils/simulation/interfaces.ts:111]: CLClientKeys represents both as an enum

**Review by dapplion — COMMENTED** [review]
Overall looks good, some minor comments

**Review by g11tech — DISMISSED** [review]
lgtm!

**Review by g11tech — APPROVED** [review]
lgtm

**Comment by dapplion** [comment]
- @nazarhussain can you rebase unstable on this branch https://github.com/ChainSafe/lodestar/pull/4774 ? The current main diff is covered in your PR

**Pull request merged**